src/datastore.jl

```diff
@@ -1,10 +1,31 @@
 ## datastore
 using Distributed
 
-struct DRef
+mutable struct DRef
     owner::Int
     id::Int
     size::UInt
+    function DRef(owner, id, size)
+        d = new(owner, id, size)
+        poolref(d)
+        finalizer(x->@async(poolunref(x)), d)
+        d
+    end
+end
+
+function Serialization.deserialize(io::AbstractSerializer, dt::Type{DRef})
+    nf = fieldcount(dt)
+    d = ccall(:jl_new_struct_uninit, Any, (Any,), dt)
+    Serialization.deserialize_cycle(io, d)
```

Comment:
would you be able to add some comments for the serialize and deserialize functions? We should make sure there's no weird interaction with the serialize function for `MMWrap`.

Reply:
These are copied from Dagger's `Chunk` deserialize implementation. All it does is manually construct the object and then attach a finalizer. I assume this is necessary since the inner constructor (and thus the finalizer callback) won't otherwise get called.